lab_1_keywords_tfidf/main.py

```diff
@@ -48,24 +71,56 @@ def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
 
     In case of corrupt input arguments, None is returned
     """
+    frequences = {}
+    if not (isinstance(tokens, list)):
+        return None
+    for token in tokens:
+        occurance_number = token.count(token)
+        frequences[token] = occurance_number
+    return frequences
     pass
 
-
-def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[list[str]]:
-    """
-    Extracts a certain number of most frequent tokens
-
-    Parameters:
-    frequencies (Dict): A dictionary with tokens and
-    its corresponding frequency values
-    top (int): Number of token to extract
-
-    Returns:
-    List[str]: Sequence of specified length
-    consisting of tokens with the largest frequency
-
-    In case of corrupt input arguments, None is returned
-    """
+    def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[list[str]]:
```

Comment:
I appreciate all the efforts you've put into this function
However, using plain and simple _sorted_ with one certain argument could do the job just as well